src/main/fc/fc_core.c

```diff
@@ -434,6 +495,66 @@ void processRx(timeUs_t currentTimeUs)
         pidStabilisationState(PID_STABILISATION_ON);
     }
 
+    // If runaway_takeoff_prevention is enabled, accumulate the amount of time that throttle
+    // is above runaway_takeoff_deactivate_throttle with the any of the R/P/Y sticks deflected
+    // to at least runaway_takeoff_stick_percent percent while the pidSum on all axis is kept low.
+    // Once the amount of accumulated time exceeds runaway_takeoff_deactivate_delay then disable
+    // prevention for the remainder of the battery.
+
+    if (ARMING_FLAG(ARMED)
+        && pidConfig()->runaway_takeoff_prevention
+        && !runawayTakeoffCheckDisabled
+        && !STATE(FIXED_WING)) {
+
+        // Determine if we're in "flight"
+        //   - motors running
+        //   - throttle over runaway_takeoff_deactivate_throttle_percent
+        //   - sticks are active and have deflection greater than runaway_takeoff_deactivate_stick_percent
+        //   - pidSum on all axis is less then runaway_takeoff_deactivate_pidlimit
+        inFlight = false;
+        if (!feature(FEATURE_MOTOR_STOP) || isAirmodeActive() || (calculateThrottleStatus() != THROTTLE_LOW)) { // are motors running?
+            float pidSumLimit = pidConfig()->runaway_takeoff_deactivate_pidlimit * 10.0f;
+            if ((calculateThrottlePercent() >= pidConfig()->runaway_takeoff_deactivate_throttle)
+                && areSticksActive(pidConfig()->runaway_takeoff_deactivate_stick)
+                && (fabsf(axisPID_P[FD_PITCH] + axisPID_I[FD_PITCH] + axisPID_D[FD_PITCH]) < pidSumLimit)
+                && (fabsf(axisPID_P[FD_ROLL] + axisPID_I[FD_ROLL] + axisPID_D[FD_ROLL]) < pidSumLimit)
+                && (fabsf(axisPID_P[FD_YAW] + axisPID_I[FD_YAW]) < pidSumLimit)) {
+                
+                inFlight = true;
+                if (runawayTakeoffDeactivateUs == 0) {
+                    runawayTakeoffDeactivateUs = currentTimeUs;
+                }
+            }
+        }
+
+        // If we're in flight, then accumulate the time and deactivate once it exceeds runaway_takeoff_deactivate_delay milliseconds
+        if (inFlight) {
+            if (runawayTakeoffDeactivateUs == 0) {
+                runawayTakeoffDeactivateUs = currentTimeUs;
+            }
+            if ((cmpTimeUs(currentTimeUs, runawayTakeoffDeactivateUs) + runawayTakeoffAccumulatedUs) > (pidConfig()->runaway_takeoff_deactivate_delay * 1000)) {
+                runawayTakeoffCheckDisabled = true;
+            }
+
+        } else {
+            if (runawayTakeoffDeactivateUs != 0) {
+                runawayTakeoffAccumulatedUs += cmpTimeUs(currentTimeUs, runawayTakeoffDeactivateUs);
```

Comment:
It seems unnecessary to accumulate time like this. Just set the time from when you started and compare with `currentTimeUs`

Reply:
That would change the way this works though. The current implementation registers a successful flight and deactivates the check even if the pidsums exceed the limit or the sticks returned to their default positions at times. If this wasn't the case, then there would be a chance that the check never got deactivated during the flight.

Reply:
OK, I didn't look into the precise detail. It just seemed a bit strange.

Reply:
Correct.  The logic needs to accumulate multiple periods where the conditions are all true with possible gaps where they're not.  Originally the logic was a simple delay based on all the conditions being true for the full duration with a reset if not.  The necessity of adding stick activity made it difficult to find an appropriate delay time as all the conditions were only intermittently true.  Either we had to have too short a delay and risk false deactivation (the main problem), or deactivation would take too long (or not happen at all) because the conditions wouldn't be met contiguously for enough time - particularly throttle and stick deflection.  With the accumulation we gather time when stable flight appears to be underway until we have enough data to be confident.  Also the transitional events like throttle changes and stick deflection are what exacerbate the runaway so having multiple events while pidSum stays low really helps.